packages/typescript/src/lib/TypescriptCompiler.ts

```diff
@@ -112,26 +101,44 @@ export class TypescriptCompiler {
       sourceFiles,
       data
     ) => {
-      // for exposes: { "./expose/path": "path/to/file" }
-      // force typescript to write compiled output to "@mf-typescript/expose/path"
-      const sourceFilename = sourceFiles?.[0].fileName || '';
-
-      // Try to rewrite the path with exposed federated modules,
-      // failing so, use the default filepath emitted by TS Compiler.
-      // This second case is valid for 'additionalFileToCompiler' added through Plugin Options.
-      const normalizedFilepath =
-        rewritePathsWithExposedFederatedModules(sourceFilename) ?? filepath;
-
-      this.tsDefinitionFilesObj[normalizedFilepath] = text;
-
+      this.tsDefinitionFilesObj[filepath] = text;
       originalWriteFile(
-        normalizedFilepath,
+        filepath,
         text,
         writeOrderByteMark,
         onError,
         sourceFiles,
         data
       );
+
+      // create exports matching the `exposes` config
+      const sourceFilename = sourceFiles?.[0].fileName || '';
```

Comment:
So, without having another `_types` folder nested, the entire `host.writeFile` function can be re-factored to pretty much this

```ts
host.writeFile = (filepath, text, writeOrderByteMark, onError, sourceFiles, data)=>{

  originalWriteFile(filepath, text, writeOrderByteMark, onError, sourceFiles, data);

  this.tsDefinitionFilesObj[filepath] = text;

  const sourceFilename = (sourceFiles == null ? void 0 : sourceFiles[0].fileName) || '';
  const normalizedFilepath = rewritePathsWithExposedFederatedModules(sourceFilename);

  // for exposes: { "./expose/path": "path/to/file" }
  // force typescript to write compiled output to "@mf-typescript/expose/path"
  if (normalizedFilepath) {
    const rp = _path.default.relative(_path.default.dirname(normalizedFilepath), filepath);
    const updatedText = `export * from './${rp.replace('.d.ts', '')}'`;

    this.tsDefinitionFilesObj[normalizedFilepath] = updatedText;

    originalWriteFile(normalizedFilepath, updatedText, writeOrderByteMark, onError, sourceFiles, data);
  }
};
```

Reply:
I agree having the nested folder is not the best, I'm concerned about outputting the rewritten path in the same dir as the initial output - there would be edge cases where the rewritten file overwrites the original and we lose the type info

```
'./components/button': './components/button'
```

We would get mf-types/components/button from the TS compiler, then it would be overwritten by the custom logic?

Reply:
Perhaps it's just the naming, maybe something along the lines of `_internal`?

Reason I didn't go for a vague name was because the consumer might still import these internal types and `remote/_types/additionalfilethatwascompiled` made more sense to me

Reply:
Just a note: the updatedText in the example works only if you don't have a default export

Reply:
@hongkiulam makes sense 👍 and I wouldn't bother about the name of the folder since we are allowing user to modify.